pallets/parachain-staking/src/lib.rs

```diff
@@ -2190,23 +2134,25 @@ pub mod pallet {
 					}
 				})?;
 
-			// update total stake
-			state.total = state
-				.total
-				.saturating_sub(stake_to_remove.amount)
-				.saturating_add(stake.amount);
-
-			// update storage of kicked delegator
-			Self::kick_delegator(&stake_to_remove, &state.id)?;
-
-			Self::deposit_event(Event::DelegationReplaced(
-				stake.owner,
-				stake.amount,
-				stake_to_remove.owner,
-				stake_to_remove.amount,
-				state.id.clone(),
-				state.total,
-			));
+			if let Some(stake_to_remove) = stake_to_remove {
+				// update total stake
+				state.total = state
+					.total
+					.saturating_sub(stake_to_remove.amount)
+					.saturating_add(stake.amount);
```

Comment:
check if this is fine. What if nobody was removed from the delegators? We don't increase the total stake?

Reply:
fixed here d404ef222601c6551ee27a7951ed788593eacbca just to be safe